From 0fa2c5243ed2d0d3c8668ef4c4282d6619a39858 Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Fri, 20 Jan 2017 12:53:52 +0800
Subject: [PATCH 1497/5242] MLK-13733-4 ARM: imx: pm-rpmsg: allign header
 format as M4 defined

commit  b018b92397abf077118749ac10eed2e248638cb2 from
https://source.codeaurora.org/external/imx/linux-imx.git

Alllign header format as M4 defined, no need revert order. Correct
the data to u8 instead of u32 as M4 defined. Call the header define
in imx_rpmsg.h derectly.

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/pm-rpmsg.c |   30 ++++++++++++++----------------
 1 file changed, 14 insertions(+), 16 deletions(-)

diff --git a/arch/arm/mach-imx/pm-rpmsg.c b/arch/arm/mach-imx/pm-rpmsg.c
index 680f213..a98bf8f 100644
--- a/arch/arm/mach-imx/pm-rpmsg.c
+++ b/arch/arm/mach-imx/pm-rpmsg.c
@@ -15,6 +15,7 @@
 
 #include <linux/err.h>
 #include <linux/slab.h>
+#include <linux/imx_rpmsg.h>
 #include <linux/interrupt.h>
 #include <linux/module.h>
 #include <linux/of.h>
@@ -26,8 +27,6 @@
 
 #define RPMSG_TIMEOUT 1000
 
-#define PM_RPMSG_LIFE_CYCLE	1
-#define PM_RPMSG_VERSION	1
 #define PM_RPMSG_TYPE		2
 
 enum pm_rpmsg_cmd {
@@ -56,12 +55,9 @@ struct pm_rpmsg_info {
 static struct delayed_work heart_beat_work;
 
 struct pm_rpmsg_data {
-	u8 cmd;
-	u8 type;
-	u8 version;
-	u8 cat;
-	u32 data;
-};
+	struct imx_rpmsg_head header;
+	u8 data;
+} __attribute__ ((packed));
 
 static int pm_send_message(struct pm_rpmsg_data *msg,
 			struct pm_rpmsg_info *info)
@@ -89,10 +85,11 @@ void pm_vlls_notify_m4(bool enter)
 {
 	struct pm_rpmsg_data msg;
 
-	msg.cat = PM_RPMSG_LIFE_CYCLE;
-	msg.version = PM_RPMSG_VERSION;
-	msg.type = PM_RPMSG_TYPE;
-	msg.cmd = PM_RPMSG_MODE;
+	msg.header.cate = IMX_RMPSG_LIFECYCLE;
+	msg.header.major = IMX_RMPSG_MAJOR;
+	msg.header.minor = IMX_RMPSG_MINOR;
+	msg.header.type = PM_RPMSG_TYPE;
+	msg.header.cmd = PM_RPMSG_MODE;
 	msg.data = enter ? PM_RPMSG_VLLS : PM_RPMSG_RUN;
 
 	pm_send_message(&msg, &pm_rpmsg);
@@ -102,10 +99,11 @@ static void pm_heart_beat_work_handler(struct work_struct *work)
 {
 	struct pm_rpmsg_data msg;
 
-	msg.cat = PM_RPMSG_LIFE_CYCLE;
-	msg.version = PM_RPMSG_VERSION;
-	msg.type = PM_RPMSG_TYPE;
-	msg.cmd = PM_RPMSG_HEART_BEAT;
+	msg.header.cate = IMX_RMPSG_LIFECYCLE;
+	msg.header.major = IMX_RMPSG_MAJOR;
+	msg.header.minor = IMX_RMPSG_MINOR;
+	msg.header.type = PM_RPMSG_TYPE;
+	msg.header.cmd = PM_RPMSG_HEART_BEAT;
 	msg.data = 0;
 
 	pm_send_message(&msg, &pm_rpmsg);
-- 
1.7.9.5

